src/Components/Web.JS/src/Platform/BootConfig.ts

```diff
@@ -1,13 +1,17 @@
+import { WebAssemblyBootResourceType } from "./WebAssemblyStartOptions";
+
 export class BootConfigResult {
   private constructor(public bootConfig: BootJsonData, public applicationEnvironment: string) {
   }
 
-  static async initAsync(environment?: string): Promise<BootConfigResult> {
-    const bootConfigResponse = await fetch('_framework/blazor.boot.json', {
-      method: 'GET',
-      credentials: 'include',
-      cache: 'no-cache'
-    });
+  static async initAsync(loadBootResource?: (type: WebAssemblyBootResourceType, name: string, defaultUri: string, integrity: string) => string | Promise<Response> | null | undefined, environment?: string): Promise<BootConfigResult> {
```

Comment:
This is a hyper-complicated type definition. Could you declare a type or interface for some of this to make it short enough to be readable?

For example for the `(type: WebAssemblyBootResourceType, name: string, defaultUri: string, integrity: string) => string | Promise<Response> | null | undefined` part